### ISSUE TITLE
Playlist auto deleting

### ISSUE THREAD

**Issue by cydia2020** [timusus/Shuttle#101]
**Shuttle version:**
 
 v1.6.6
 
**Device, OS:**
 
Google Pixel XL
[PureNexusProject](https://github.com/cydia2020)-Android 7.1.2

**Description of bug:**
 
Playlist auto delete by itself. Happening randomly
 
**Steps to reproduce:**
 
 1. Add a playlist
 2. Listen to music for few days
 3. Playlist disappears
 
**Expected outcome:**
 
Playlist should always stay

**Comment by timusus** [comment]
Lots of reports of this same issue. I'm not convinced that Shuttle is responsible for these playlists disappearing, but I don't know what the cause is.

**Comment by cydia2020** [comment]
Yes, this even happens on Google Play Music, but maybe if Shuttle is using its own playlist system, then this sorta things won't happen just like Poweramp. And from what I know this seems to be a problem that only happens on a near AOSP system, costumed Android OS like HTC Sense doesn't have these kind of problem.

**Comment by timusus** [comment]
One of the main problems with implementing a custom playlist system, is third party apps (like Zombies, Run) rely on playlists being available in the MediaStore and actually recommend their users download Shuttle because it works with MediaStore playlists.

I'd rather figure out what the root cause of this issue is, if possible, than write a workaround for a firmware issue.

**Comment by cydia2020** [comment]
OK, thanks for that! 👍

**Comment by timusus** [comment]
Not getting anywhere in identifying the cause of this. It would be good to implement a custom playlist solution.. Perhaps storing playlists in a local DB.

**Comment by Obap** [comment]
For me I have a clear cause playlists disappear when there is an update.

**Comment by FichteFoll** [comment]
Vanilla Music recently added a feature to sync Playlist as m3u files in a /Playlists folder. It's still in development, however. Maybe shuttle could make use of this as well?

https://github.com/vanilla-music/vanilla/issues/643
